### PR TITLE
[core] Fold AutoloadLibraryMU into AutoloadLibraryGenerator

### PR DESCRIPTION
**Pull request by hahnjo**
Use `AbsoluteSymbolsMaterializationUnit` to define the symbols.

### CODE REVIEW

**Comment by hahnjo** [comment]
As far as I could understand, Lang recommends this simplification, so marking as ready for review

File: core/metacling/src/TClingCallbacks.cxx [hunk 1]
```diff
@@ -200,10 +128,27 @@ class AutoloadLibraryGenerator : public llvm::orc::DefinitionGenerator {
...
       }
 
-      for (auto &&KV : found) {
-         auto MU = std::make_unique<AutoloadLibraryMU>(KV.first, std::move(KV.second));
-         if (auto Err = JD.define(MU))
-            return Err;
+      llvm::orc::SymbolMap loadedSymbols;
+      for (const auto &KV : found) {
+         // Try to load the library which should provide the symbol definition.
+         // TODO: Should this interface with the DynamicLibraryManager directly?
```
Comment (vgvassilev) [inline core/metacling/src/TClingCallbacks.cxx:134]: Yes, ideally that'd be a good idea...

**Comment by hahnjo** [comment]
> Do we have any performance numbers? Is the absolute symbol approach similar in terms of computational cost?

I'm not sure I understand, this PR is just moving the code from the MU to the Generator. `AbsoluteSymbolsMaterializationUnit` is literally just doing a `notifyResolved` of the symbols with their absolute pointers.

**Comment by vgvassilev** [comment]
> > Do we have any performance numbers? Is the absolute symbol approach similar in terms of computational cost?
> 
> I'm not sure I understand, this PR is just moving the code from the MU to the Generator. `AbsoluteSymbolsMaterializationUnit` is literally just doing a `notifyResolved` of the symbols with their absolute pointers.

I think it answers my question.

**Review by vgvassilev — APPROVED** [review]
Lgtm!

**Pull request merged**